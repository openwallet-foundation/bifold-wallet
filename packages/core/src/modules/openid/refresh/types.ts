import {
  ClaimFormat,
  JwaSignatureAlgorithm,
  JwkJson,
  MdocRecord,
  SdJwtVcRecord,
  W3cCredentialRecord,
} from '@credo-ts/core'
import { OpenId4VciResolvedCredentialOffer } from '@credo-ts/openid4vc'

export type IssuerMetadataCache = {
  credential_issuer: string
  credential_endpoint?: string
  token_endpoint?: string
  authorization_servers?: string[]
  credential_configurations_supported?: Record<string, any>
  // optionally: added_at, etag, updated_at, etc.
}

export enum RefreshStatus {
  Valid = 'valid',
  Invalid = 'invalid',
  Error = 'error',
}

export interface RefreshCredentialMetadata {
  // issuer & config to re-issue the same cred
  authServer: string
  tokenEndpoint: string
  refreshToken: string
  credentialIssuer: string
  credentialConfigurationId: string
  issuerMetadataCache: IssuerMetadataCache
  clientId?: string
  tokenBinding?: 'DPoP' | 'Bearer'
  dpop?: { alg: JwaSignatureAlgorithm; jwk: JwkJson }

  // refresh lifecycle
  lastCheckedAt?: number
  lastCheckResult?: RefreshStatus
  attemptCount?: number
  nextEligibleAt?: number

  // linking (optional)
  supersededByID?: string
  supersedesID?: string
  resolvedCredentialOffer?: OpenId4VciResolvedCredentialOffer
}

export type RefreshOrchestratorOpts = {
  intervalMs?: number | null
  autoStart?: boolean
  onError?: (e: unknown) => void
  listRecords?: () => Promise<any[]>
  toLite?: (rec: W3cCredentialRecord | SdJwtVcRecord | MdocRecord) => {
    id: string
    format: ClaimFormat
    createdAt?: string
    issuer?: string
  }
}

export interface IRefreshOrchestrator {
  configure(opts: Partial<RefreshOrchestratorOpts>): void
  start(): void
  stop(): void
  runOnce(reason?: string): Promise<void>
  isRunning(): boolean
  resolveFull(id: string): W3cCredentialRecord | SdJwtVcRecord | MdocRecord | undefined
}

export enum OpenIDCustomNotificationType {
  CredentialReplacementAvailable = 'CustomNotificationOpenIDCredential',
}
