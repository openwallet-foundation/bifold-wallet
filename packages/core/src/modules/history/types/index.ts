import { GenericRecord } from '@credo-ts/core/build/modules/generic-records/repository/GenericRecord'
import { Moment } from 'moment'

export interface IHistoryManager {
  saveHistory: (recordData: HistoryRecord) => Promise<void>
  getHistoryItems: (query: HistoryQuery) => Promise<CustomRecord[]>
  findGenericRecordById: (id: string) => Promise<GenericRecord | null>
  removeGenericRecord: (genericRecord: GenericRecord) => Promise<void>
  handleStoreHistorySettings: (selectedValue: HistoryBlockSelection | undefined) => Promise<void>
  getStoredHistorySettingsOption: () => Promise<string | null>
  getHistorySettingsOptionList: () => Array<HistoryBlockSelection>
}

export interface HistoryBlockSelection {
  value: string
  id: string
  key?: number
  date?: Moment
}

export enum HistoryCardType {
  CardAccepted = 'CardAccepted',
  CardDeclined = 'CardDeclined',
  CardExpired = 'CardExpired', // TODO: log this type of event
  CardRemoved = 'CardRemoved',
  CardRevoked = 'CardRevoked', // TODO: log this type of event
  CardUpdates = 'CardUpdates', // TODO: log this type of event
  PinChanged = 'PinChanged',
  InformationSent = 'InformationSent',
  InformationNotSent = 'InformationNotSent',
  Connection = 'Connection',
  ConnectionRemoved = 'ConnectionRemoved',
  ActivateBiometry = 'ActivateBiometry',
  DeactivateBiometry = 'DeactivateBiometry',
}
export interface HistoryRecord {
  /** History Record ID */
  id?: string
  /** History Record Type */
  type?: HistoryCardType
  /** History Record Message */
  message: string
  /** History Record Date */
  createdAt?: Date
  /** History Record Item ID */
  correspondenceId?: string
  /** History Record Item Name */
  correspondenceName?: string
}
export enum RecordType {
  AppNotificationRecord = 'AppNotificationRecord',
  NotificationRecord = 'NotificationRecord',
  HistoryRecord = 'HistoryRecord',
}
export type TagObject = {
  [key: string]: string | boolean | undefined | Array<string> | null
  [key: number]: never
}

export interface CustomRecord {
  content: RecordTypes
  tags?: TagObject
}

export interface AppNotificationRecord {
  id?: string
  createdAt?: Date
  type?: AppNotificationCardType
}

export interface NotificationRecord {
  message: string
  credentialId: string
  type?: NotificationCardType
  id?: string
  createdAt?: Date
  isNotificationRemoved: boolean
}

export enum AppNotificationCardType {
  AppPinReminder = 'AppPinReminder',
}

export enum NotificationCardType {
  CardExpired = 'CardExpired',
  CardRevoked = 'CardRevoked',
}

type RecordTypes = NotificationRecord | HistoryRecord | AppNotificationRecord
export type HistoryQuery = {
  type: RecordType
}
