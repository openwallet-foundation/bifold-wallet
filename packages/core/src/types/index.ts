import * as state from './state'
export { state }
