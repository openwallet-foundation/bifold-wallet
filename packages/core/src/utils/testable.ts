import { testIdPrefix } from '../constants'

export const testIdWithKey = (key: string): string => {
  return `${testIdPrefix}${key}`
}

export const testIdForAccessabilityLabel = (label: string) => {
  if (!label) {
    return ''
  }

  return label
    .split(' ')
    .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
    .join(' ')
    .replace(/\s/g, '')
}
