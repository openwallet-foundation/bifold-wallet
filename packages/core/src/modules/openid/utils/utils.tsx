import { getDomainFromUrl } from '@credo-ts/core'
import { Attribute, Field } from '@hyperledger/aries-oca/build/legacy'
import { W3cCredentialDisplay } from '../types'

/**
 * Converts a camelCase string to a sentence format (first letter capitalized, rest in lower case).
 * i.e. sanitizeString("helloWorld")  // returns: 'Hello world'
 */
export function sanitizeString(str: string) {
  const result = str.replace(/([a-z0-9])([A-Z])/g, '$1 $2').replaceAll('_', ' ')
  let words = result.split(' ')
  words = words.map((word, index) => {
    if (index === 0 || word.toUpperCase() === word) {
      return word.charAt(0).toUpperCase() + word.slice(1)
    }
    return word.charAt(0).toLowerCase() + word.slice(1)
  })
  return words.join(' ')
}

export function getHostNameFromUrl(url: string) {
  try {
    return getDomainFromUrl(url)
  } catch (error) {
    throw new Error(`Error getting hostname from url: ${error}`)
  }
}

export const buildFieldsFromOpenIDTemplate = (data: { [key: string]: unknown }): Array<Field> => {
  const fields = []
  for (const key of Object.keys(data)) {
    // omit id and type
    if (key === 'id' || key === 'type') continue

    let pushedVal: string | number | null = null
    if (typeof data[key] === 'string' || typeof data[key] === 'number') {
      pushedVal = data[key] as string | number | null
    }
    fields.push(new Attribute({ name: key, value: pushedVal }))
  }
  return fields
}

export function formatDate(input: string | Date): string {
  const date = input instanceof Date ? input : new Date(input)
  return date.toLocaleString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit',
    hour12: true,
  })
}

export const getAttributeField = (display: W3cCredentialDisplay, searchKey: string): Attribute | undefined => {
  for (const [key, value] of Object.entries(display.attributes)) {
    if (searchKey === key) {
      return new Attribute({
        name: key,
        value: value as string | number | null,
        mimeType: typeof value === 'number' ? 'text/number' : 'text/plain',
      })
    }
  }
  return undefined
}

/**
 * Mostly used for mdoc crednetials
 */
export function detectImageMimeType(data: Uint8Array): 'image/jpeg' | 'image/jp2' | null {
  // Check if array is too short to contain magic numbers
  if (data.length < 12) {
    return null
  }

  // JPEG starts with FF D8 FF
  if (data[0] === 0xff && data[1] === 0xd8 && data[2] === 0xff) {
    return 'image/jpeg'
  }

  // JPEG2000 has two possible signatures:
  // 1) 00 00 00 0C 6A 50 20 20 0D 0A 87 0A
  // 2) FF 4F FF 51

  // Check first signature
  if (
    data[0] === 0x00 &&
    data[1] === 0x00 &&
    data[2] === 0x00 &&
    data[3] === 0x0c &&
    data[4] === 0x6a && // 'j'
    data[5] === 0x50 && // 'P'
    data[6] === 0x20 &&
    data[7] === 0x20 &&
    data[8] === 0x0d &&
    data[9] === 0x0a &&
    data[10] === 0x87 &&
    data[11] === 0x0a
  ) {
    return 'image/jp2'
  }

  // Check second signature
  if (data[0] === 0xff && data[1] === 0x4f && data[2] === 0xff && data[3] === 0x51) {
    return 'image/jp2'
  }

  return null
}

/**
 * very simple matcher for `yyyy-mm-dd`
 */
export function isDateString(value: string) {
  // We do the length check first to avoid unnecesary regex
  return value.length === 'yyyy-mm-dd'.length && value.match(/^\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01])$/)
}

export function isW3CProofRequest(type: string): boolean {
  if (type === 'jwt_vc') {
    return true
  }
  return false
}

export function isSdJwtProofRequest(type: string): boolean {
  if (type === 'vc+sd-jwt') {
    return true
  }
  return false
}
