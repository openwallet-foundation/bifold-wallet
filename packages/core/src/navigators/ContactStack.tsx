import { createStackNavigator } from '@react-navigation/stack'
import React from 'react'
import { useTranslation } from 'react-i18next'

import HeaderRightHome from '../components/buttons/HeaderHome'
import { useTheme } from '../contexts/theme'
import Chat from '../screens/Chat'
import ContactDetails from '../screens/ContactDetails'
import CredentialDetails from '../screens/CredentialDetails'
import CredentialOffer from '../screens/CredentialOffer'
import ListContacts from '../screens/ListContacts'
import ProofDetails from '../screens/ProofDetails'
import ProofRequest from '../screens/ProofRequest'
import RenameContact from '../screens/RenameContact'
import WhatAreContacts from '../screens/WhatAreContacts'
import { ContactStackParams, Screens } from '../types/navigators'

import { useDefaultStackOptions } from './defaultStackOptions'
import { TOKENS, useServices } from '../container-api'

const ContactStack: React.FC = () => {
  const Stack = createStackNavigator<ContactStackParams>()
  const theme = useTheme()
  const { t } = useTranslation()
  const defaultStackOptions = useDefaultStackOptions(theme)
  const [ScreenOptionsDictionary] = useServices([TOKENS.OBJECT_SCREEN_CONFIG])

  return (
    <Stack.Navigator screenOptions={{ ...defaultStackOptions }}>
      <Stack.Screen
        name={Screens.Contacts}
        component={ListContacts}
        options={{ title: t('Screens.Contacts'), ...ScreenOptionsDictionary[Screens.Contacts] }}
      />
      <Stack.Screen
        name={Screens.ContactDetails}
        component={ContactDetails}
        options={{
          title: t('Screens.ContactDetails'),
          ...ScreenOptionsDictionary[Screens.ContactDetails],
        }}
      />
      <Stack.Screen
        name={Screens.RenameContact}
        component={RenameContact}
        options={{
          title: t('Screens.RenameContact'),
          ...ScreenOptionsDictionary[Screens.RenameContact],
        }}
      />
      <Stack.Screen
        name={Screens.Chat}
        component={Chat}
        options={{
          ...ScreenOptionsDictionary[Screens.Chat],
        }}
      />
      <Stack.Screen
        name={Screens.WhatAreContacts}
        component={WhatAreContacts}
        options={{
          title: '',
          ...ScreenOptionsDictionary[Screens.WhatAreContacts],
        }}
      />
      <Stack.Screen
        name={Screens.CredentialDetails}
        component={CredentialDetails}
        options={{ title: t('Screens.CredentialDetails'), ...ScreenOptionsDictionary[Screens.CredentialDetails] }}
      />
      <Stack.Screen
        name={Screens.CredentialOffer}
        component={CredentialOffer}
        options={{ title: t('Screens.CredentialOffer'), ...ScreenOptionsDictionary[Screens.CredentialOffer] }}
      />
      <Stack.Screen
        name={Screens.ProofDetails}
        component={ProofDetails}
        options={() => ({
          title: '',
          headerRight: () => <HeaderRightHome />,
          ...ScreenOptionsDictionary[Screens.ProofDetails],
        })}
      />
      <Stack.Screen
        name={Screens.ProofRequest}
        component={ProofRequest}
        options={{
          title: t('Screens.ProofRequest'),
          ...ScreenOptionsDictionary[Screens.ProofRequest],
        }}
      />
    </Stack.Navigator>
  )
}

export default ContactStack
