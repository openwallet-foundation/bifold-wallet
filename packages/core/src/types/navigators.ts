import { MdocRecord, SdJwtVcRecord, W3cCredentialRecord } from '@credo-ts/core'
import { NavigatorScreenParams } from '@react-navigation/native'
import { StackNavigationOptions } from '@react-navigation/stack'
import { OpenId4VPRequestRecord, OpenIDCredentialType } from '../modules/openid/types'
import { LayoutProps } from '../layout/ScreenLayout'

export enum Screens {
  AttemptLockout = 'Temporarily Locked',
  Splash = 'Splash',
  Onboarding = 'Onboarding',
  Terms = 'Terms',
  Preface = 'Preface',
  CreatePIN = 'Create a PIN',
  ChangePIN = 'Change PIN',
  EnterPIN = 'Enter PIN',
  Home = 'Home',
  Scan = 'Scan',
  PasteUrl = 'Paste URL',
  Credentials = 'Credentials',
  CredentialDetails = 'Credential Details',
  CredentialOffer = 'Credential Offer',
  OpenIDCredentialDetails = 'Open ID Credential details',
  OpenIDCredentialOffer = 'Open ID Credential offer',
  OpenIDProofPresentation = 'Open ID Proof Presentation',
  OpenIDProofCredentialSelect = 'Open ID Proof Credential Select',
  ProofRequest = 'Proof Request',
  ProofRequestDetails = 'Proof Request Details',
  ProofRequestUsageHistory = 'Proof Request Usage History',
  Settings = 'Settings',
  Language = 'Language',
  DataRetention = 'Data Retention',
  Tours = 'Tours',
  Contacts = 'Contacts',
  ContactDetails = 'Contact Details',
  WhatAreContacts = 'What Are Contacts',
  Chat = 'Chat',
  Connection = 'Connection',
  MobileVerifierLoading = 'Mobile Verifier Loading',
  OnTheWay = 'On The Way',
  Declined = 'Declined',
  Biometry = 'Biometry',
  ToggleBiometry = 'Toggle Biometry',
  PushNotifications = 'Push Notifications',
  TogglePushNotifications = 'Toggle Push Notifications',
  Developer = 'Developer',
  CustomNotification = 'Custom Notification',
  ProofChangeCredential = 'Choose a credential',
  ProofRequests = 'Proof Requests',
  ProofRequesting = 'Proof Requesting',
  ProofDetails = 'Proof Details',
  NameWallet = 'Name Wallet',
  RenameWallet = 'Rename Wallet',
  RenameContact = 'Rename Contact',
  JSONDetails = 'JSON Details',
  ScanHelp = 'Scan Help',
  HistorySettings = 'History Settings',
  HistoryPage = 'History',
  HistoryDetails = 'History details',
  AutoLock = 'AutoLock',
  UpdateAvailable = 'Update Available',
  ConfigureMediator = 'Configure Mediator',
}

export enum Stacks {
  TabStack = 'Tab Stack',
  HomeStack = 'Home Stack',
  ConnectStack = 'Connect Stack',
  CredentialStack = 'Credentials Stack',
  SettingStack = 'Settings Stack',
  ContactStack = 'Contacts Stack',
  ProofRequestsStack = 'Proof Requests Stack',
  NotificationStack = 'Notifications Stack',
  ConnectionStack = 'Connection Stack',
  HistoryStack = 'History Stack',
  CustomNavStack1 = 'Custom Nav Stack 1',
}

export enum TabStacks {
  HomeStack = 'Tab Home Stack',
  ConnectStack = 'Tab Connect Stack',
  CredentialStack = 'Tab Credential Stack',
}

export type RootStackParams = {
  [Screens.Splash]: undefined
  [Stacks.TabStack]: NavigatorScreenParams<TabStackParams>
  [Screens.Chat]: { connectionId: string }
  [Stacks.ConnectStack]: NavigatorScreenParams<ConnectStackParams>
  [Stacks.ConnectionStack]: NavigatorScreenParams<DeliveryStackParams>
  [Stacks.SettingStack]: NavigatorScreenParams<SettingStackParams>
  [Stacks.ContactStack]: NavigatorScreenParams<ContactStackParams>
  [Stacks.ProofRequestsStack]: NavigatorScreenParams<ProofRequestsStackParams>
  [Stacks.NotificationStack]: NavigatorScreenParams<NotificationStackParams>
  [Stacks.HistoryStack]: NavigatorScreenParams<HistoryStackParams>
  [Screens.CredentialDetails]: { credentialId: string }
  [Screens.OpenIDCredentialDetails]: { credentialId: string; type: OpenIDCredentialType }
  [Stacks.CustomNavStack1]: undefined
}

export type TabStackParams = {
  [TabStacks.HomeStack]: NavigatorScreenParams<HomeStackParams>
  [TabStacks.ConnectStack]: NavigatorScreenParams<ConnectStackParams>
  [TabStacks.CredentialStack]: NavigatorScreenParams<CredentialStackParams>
}

export type OnboardingStackParams = {
  [Screens.Preface]: undefined
  [Screens.Onboarding]: undefined
  [Screens.Terms]: undefined
  [Screens.AttemptLockout]: undefined
  [Screens.CreatePIN]: { setAuthenticated: (status: boolean) => void } | undefined
  [Screens.EnterPIN]: { setAuthenticated: (status: boolean) => void } | undefined
  [Screens.Biometry]: undefined
  [Screens.NameWallet]: undefined
  [Screens.PushNotifications]: undefined
}

export type ContactStackParams = {
  [Screens.Contacts]: undefined
  [Screens.Chat]: { connectionId: string }
  [Screens.ContactDetails]: { connectionId: string }
  [Screens.RenameContact]: { connectionId: string }
  [Screens.WhatAreContacts]: undefined
  [Screens.CredentialDetails]: { credentialId: string }
  [Screens.CredentialOffer]: { credentialId: string }
  [Screens.ProofDetails]: { recordId: string; isHistory?: boolean }
  [Screens.ProofRequest]: { proofId: string }
  [Screens.JSONDetails]: { jsonBlob: string }
}

export type ProofRequestsStackParams = {
  [Screens.ProofRequests]: { connectionId?: string }
  [Screens.ProofRequesting]: { templateId: string; predicateValues?: Record<string, Record<string, number>> }
  [Screens.ProofDetails]: { recordId: string; isHistory?: boolean; senderReview?: boolean }
  [Screens.ProofRequestDetails]: { templateId: string; connectionId?: string }
  [Screens.MobileVerifierLoading]: { proofId: string; connectionId: string }
  [Screens.ProofRequestUsageHistory]: { templateId: string }
  [Screens.ProofChangeCredential]: {
    selectedCred: string
    altCredentials: string[]
    proofId: string
    onCredChange: (arg: string) => void
  }
}

export type CredentialStackParams = {
  [Screens.Credentials]: undefined
  [Screens.JSONDetails]: { jsonBlob: string }
}

export type HomeStackParams = {
  [Screens.Home]: undefined
}

export type ConnectStackParams = {
  [Screens.Scan]: undefined
  [Screens.RenameWallet]: undefined
  [Screens.ScanHelp]: undefined
  [Screens.PasteUrl]: undefined
}

export type SettingStackParams = {
  [Screens.RenameWallet]: undefined
  [Screens.Settings]: undefined
  [Screens.Language]: undefined
  [Screens.DataRetention]: undefined
  [Screens.Tours]: undefined
  [Screens.ToggleBiometry]: undefined
  [Screens.ChangePIN]: undefined
  [Screens.Terms]: undefined
  [Screens.Onboarding]: undefined
  [Screens.Developer]: undefined
  [Screens.TogglePushNotifications]: undefined
  [Screens.HistorySettings]: undefined
  [Screens.AutoLock]: undefined
  [Screens.ConfigureMediator]: { scannedMediatorUri: string } | undefined
}

export type NotificationStackParams = {
  [Screens.CredentialDetails]: { credentialId: string }
  [Screens.OpenIDCredentialDetails]: {
    credential: SdJwtVcRecord | W3cCredentialRecord
  }
  [Screens.CredentialOffer]: { credentialId: string }
  [Screens.ProofRequest]: { proofId: string }
  [Screens.CustomNotification]: undefined
  [Screens.ProofDetails]: { recordId: string }
}

export type DeliveryStackParams = {
  [Screens.Connection]: {
    oobRecordId?: string
    proofId?: string
    credentialId?: string
    openIDUri?: string
    openIDPresentationUri?: string
  }
  [Screens.MobileVerifierLoading]: { proofId: string; connectionId: string }
  [Screens.ProofDetails]: { recordId: string }
  [Screens.CredentialOffer]: { credentialId: string }
  [Screens.ProofRequest]: { proofId: string }
  [Screens.OnTheWay]: { credentialId: string }
  [Screens.Declined]: { credentialId: string }
  [Screens.Chat]: { connectionId: string }
  [Screens.OpenIDCredentialOffer]: {
    credential: SdJwtVcRecord | W3cCredentialRecord | MdocRecord
  }
  [Screens.OpenIDProofPresentation]: { credential: OpenId4VPRequestRecord }
  [Screens.OpenIDProofCredentialSelect]: {
    inputDescriptorID: string
    selectedCredID: string
    altCredIDs: {
      id: string
      claimFormat: string
    }[]
    onCredChange: ({
      inputDescriptorID,
      id,
      claimFormat,
    }: {
      inputDescriptorID: string
      id: string
      claimFormat: string
    }) => void
  }
}

export type HistoryStackParams = {
  [Screens.HistoryPage]: undefined
}

export type ScreenLayoutConfig = Partial<Record<Screens, LayoutProps>>

export type ScreenOptionsType = Partial<Record<Screens, StackNavigationOptions>>

export type OnboardingTask = {
  name: Screens
  completed: boolean
}
