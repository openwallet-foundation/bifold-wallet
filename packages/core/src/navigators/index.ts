import ConnectStack from './ConnectStack'
import ContactStack from './ContactStack'
import CredentialStack from './CredentialStack'
import DeliveryStack from './DeliveryStack'
import HomeStack from './HomeStack'
import NotificationStack from './NotificationStack'
import OnboardingStack from './OnboardingStack'
import ProofRequestStack from './ProofRequestStack'
import RootStack from './RootStack'
import SettingStack from './SettingStack'
import TabStack from './TabStack'
import { useDefaultStackOptions } from './defaultStackOptions'

export {
  ConnectStack,
  ContactStack,
  CredentialStack,
  DeliveryStack,
  HomeStack,
  NotificationStack,
  OnboardingStack,
  ProofRequestStack,
  RootStack,
  SettingStack,
  TabStack,
  useDefaultStackOptions,
}
