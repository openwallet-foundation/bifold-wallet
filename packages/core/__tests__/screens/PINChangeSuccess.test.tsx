import React from 'react'
import { render, fireEvent, act } from '@testing-library/react-native'

import ChangePINSuccessScreen from '../../src/screens/PINChangeSuccess'
import { BasicAppContext } from '../helpers/app'
import { testIdWithKey } from '../../src/utils/testable'
import { Screens } from '../../src/types/navigators'

describe('ChangePINSuccess Screen', () => {
  const mockNavigate = jest.fn()
  const mockNavigation = {
    getParent: () => ({
      navigate: mockNavigate,
    }),
    navigate: mockNavigate,
  } as any

  test('Renders correctly', async () => {
    const tree = render(
      <BasicAppContext>
        <ChangePINSuccessScreen route={{} as any} navigation={jest.fn() as any} />
      </BasicAppContext>
    )
    expect(tree).toMatchSnapshot()
  })

  test('Navigate is called with Settings Screen on "Go To Settings" press', async () => {
    const tree = render(
      <BasicAppContext>
        <ChangePINSuccessScreen route={{} as any} navigation={mockNavigation} />
      </BasicAppContext>
    )

    const primaryCTA = tree.getByTestId(testIdWithKey('GoToSettings'))

    await act(async () => {
      fireEvent.press(primaryCTA)
    })

    expect(mockNavigate).toHaveBeenCalledWith(Screens.Settings)
  })
})
