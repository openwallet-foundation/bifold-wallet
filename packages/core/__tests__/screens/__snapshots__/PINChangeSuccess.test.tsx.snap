// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePINSuccess Screen Renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Object {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "off",
    }
  }
  style={
    Object {
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    keyboardShouldPersistTaps="handled"
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#000000",
            "flex": 1,
            "padding": 20,
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 30,
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#2E8540",
                "borderRadius": 5,
                "borderWidth": 1,
                "minWidth": 700,
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "paddingHorizontal": 5,
                  "paddingTop": 5,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "marginRight": 10,
                    },
                    Object {
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  accessible={false}
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#2E8540",
                        "fontSize": 30,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  î¡¬
                </Text>
                <Text
                  maxFontSizeMultiplier={2}
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "bold",
                      },
                      Object {
                        "alignSelf": "center",
                        "color": "#FFFFFF",
                        "flexShrink": 1,
                        "marginLeft": 7,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/HeaderText"
                >
                  PINChangeSuccess.InfoBox.Title
                </Text>
              </View>
            </View>
            <RCTScrollView
              style={
                Object {
                  "flexDirection": "column",
                  "flexGrow": 0,
                  "marginLeft": 40,
                  "paddingBottom": 5,
                  "paddingHorizontal": 5,
                }
              }
            >
              <View>
                <Text
                  maxFontSizeMultiplier={2}
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "normal",
                      },
                      Object {
                        "color": "#FFFFFF",
                        "flexShrink": 1,
                        "marginVertical": 16,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/BodyText"
                >
                  PINChangeSuccess.InfoBox.Description
                </Text>
              </View>
            </RCTScrollView>
          </View>
        </View>
        <View
          accessibilityLabel="PINChangeSuccess.PrimaryButton"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#42803E",
              "borderRadius": 4,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/GoToSettings"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    },
                    false,
                    false,
                    false,
                  ],
                ]
              }
            >
              PINChangeSuccess.PrimaryButton
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
